core/app/models/spree/menu_item.rb

```diff
@@ -71,7 +71,7 @@ def frontend_available?
 
     def ensure_item_belongs_to_root
       if menu.try(:root).present? && parent_id.nil?
-        self.parent_id = menu.root.id
```

Comment:
I was testing this out and found an issue where: if  `self.parent = menu.root` is executed, awesome_nested_set doesn't register the item properly.

```ruby
    def ensure_item_belongs_to_root
      if menu.try(:root).present? && parent_id.nil?
        self.parent = menu.root
      end

      yield

      move_to_child_of(menu.root) unless root
    end
```

Reply:
Example of strange bug, 
https://www.youtube.com/watch?v=sf15t27AZXE

Reply:
Added in PR: #11028